### ISSUE TITLE
[ENH] extension templates for neural network classes

### ISSUE THREAD

**Issue by fkiraly** [sktime/pytorch-forecasting#1953]
As suggested in a recent tech meeting, we should add extension templates for the v1 networks and network packages API, to enable users to add new networks and get them tested immediately in the CI.

This could be modelled upon the extension template in `sktime`
https://github.com/sktime/sktime/tree/main/extension_templates

although, unlike in `sktime`, adding a neural network will mean adding at least two files - a `lightning` model, and a package class with tags. Thus, the extension template should probably be not a single file, but a folder, which mimicks the structure of the folder within the `models` module.

**Comment by fkiraly** [comment]
I think this is a good first issue, though I recommend getting in touch on dev-chat (discord) for anyone who has not been somewhat following the recent rework towards version 2.
https://discord.com/invite/54ACzaFsn7

**Comment by fkiraly** [comment]
One issue I just encountered, when is `get_train_test_params` necessary, when is `get_base_test_params` necessary instead? This seems to depend on which tags are populated, e.g., `"info:pred_type"`.

We should simplify the extension contract here, this is hard to understand.
FYI @phoeenniixx, @PranavBhatP, @agobbifbk, opinions appreciated!

**Comment by phoeenniixx** [comment]
The `get_base_test_params` are used to generate "Base-params" that would be paired with all the losses the model is compatible with. The tags like `"info:pred_type"` are used to get these losses. To see if the model could do classification tasks as well or not. 
So, the tags just tell us the losses models are compatible with, and the base-params would work for all these losses. This saves the effort of adding loss to each param as each param is tested against ever compatible loss, to make sure there are no "special errors" like the `LogNormalDistributionLoss` error we encountered as mentioned in #1921

**Comment by phoeenniixx** [comment]
The main downside to this structure, that I've been thinking about for a while now, is that as we add more models, the number of tests will increase very fastly (each model would have `base_params`*`all_compatible_losses` numbers of fixtures). We should think of a better way, like do this "all loss tests" for only the file changed, and for rest of the models, remaining unchanged, should get tested using something like `get_train_test_params`, that would have a smaller fixture pool with losses - not all params being tested with each loss...

**Comment by MAVRICK-1** [comment]
@fkiraly can i work on this ?

**Comment by phoeenniixx** [comment]
Yes, thank you!
You can go ahead and give it a try with v1 for now, since v2 isn’t quite mature yet.
As this isn’t a good first issue, could you please outline your approach or plan for tackling this issue?
It would be helpful to understand how you’re thinking of proceeding.

**Comment by Saptami191** [comment]
Thanks for the clarification.

From analyzing the current models, it appears that there are two sets of models in v1: simple single-file models (such as Baseline), and folder-based models with a package container that provides tags and test hooks
 (such as RecurrentNetwork).
This is confusing for people who want to contribute a new neural network model, as they would not know what structure to use and which methods are required.
Proposed solution:
1. Create a new template for a folder-based extension in models/extension_templates/
2. The new template would contain: a minimal Lightning model template (model.py),a minimal package container template (package.py) that defines required tags and a get_cls() method, with comments on which methods are optional vs required.
3. The new template would be kept very minimal and point to advanced hooks (such as get_base_test_params, custom dataloaders) as optional extensions, not part of the base template.
4. This proposal would focus on v1 and not modify v2 or any existing models.

happy to adjust the approach based on feedback before implementing.

**Comment by phoeenniixx** [comment]
Thanks a lot @Saptami191!
The solution seems good! Can you please raise a PR for this ?

**Comment by Saptami191** [comment]
Yeah I'm working on it

**Comment by harshsomankar123-tech** [comment]
@phoeenniixx If no one is currently working on this issue, can I take it up? Thanks!
I followed the proposed approach of creating a folder-based extension template for v1, https://github.com/sktime/pytorch-forecasting/issues/1953#issuecomment-3829121058 with a few adjustments:

- Location & Naming: I placed the template in a root-level extension_templates/v1_network_template/ folder (to avoid auto-registration issues) and named the package _examplenetwork_pkg.py to match the project's private package conventions.
- Test Fixtures: I included get_base_test_params and _get_test_dataloaders_from directly in the template package since they are required for CI discovery and test registration.
- Optional Overrides Boilerplate: I added commented-out templates for optional overrides like to_prediction() and to_quantiles(). This gives contributors a direct template to follow if they need custom post-processing, instead of just telling them these functions exist.

**Comment by phoeenniixx** [comment]
Thanks a lot @harshsomankar123-tech, can you please raise a PR for this?
Thanks for contributing!
